components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc

```diff
@@ -19,23 +19,23 @@ The following formula describes the overall fee for a transaction:
 
 [stem]
 ++++
-F = \text{gas_price}\cdot\left(\max_k v_k w_k + 0.9\cdot\text{calldata_cost}\cdot\left((n+2m-1) + 3t + \sum\limits_{i=1}^t q_i +3\ell\right)+ t\cdot\text{storage_write_cost} + n \cdot 200 + 272\right)
+F = \text{gas_price}\cdot\left(\max_k v_k w_k + \text{calldata_cost}\cdot\left(2n+2(m-2) + 3t + \sum\limits_{i=1}^t q_i +3\ell\right) - \text{contract_update_discount}\cdot n + 310 + \text{storage_write_cost}\cdot t\right)
 ++++
 
 where:
 
-* stem:[$v$] is a vector that represents resource usage, where each of its entries, stem:[$v_k$], correspond to different resource types: Cairo steps and number of applications of each builtin.
+* stem:[$v$] is a vector that represents resource usage, where each of its entries, stem:[$v_k$], corresponds to different resource types: Cairo steps and number of applications of each builtin.
```

Comment:
I asked two folks to take a look at this, and one of them didn't understand what k is. The other person explained that _k_ is a number that corresponds to one of the resources listed in Table 1, so k is a value 1-8, (or 0-7?). Which means that the table should have a column for resource identifier _k_.